src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1913,6 +1913,12 @@ public ErrorDiagnostic IndexOutOfBounds(string typeName, long tupleLength, long
                 DiagnosticLevel.Warning,
                 "BCP335",
                 $"The provided value {(sourceMaxLength.HasValue ? $"may have a length as large as {sourceMaxLength.Value}" : "has no configured maximum length")} and may be too long to assign to a target with a configured maximum length of {targetMaxLength}.");
+
+            public ErrorDiagnostic ParametersFileUnsupported() => new(
```

Comment:
is the reason why we switched to a diagnostic (from a CLI only error) here is to show the user the error as soon as they open up a `.bicepparam` file?

Reply:
Yes - because I wanted to block it in the LanguageServer as well, and this was the simplest option. It felt unnecessary to duplicate the logic.